plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundPartition.java

```diff
@@ -38,101 +36,73 @@
 import edu.berkeley.ground.exceptions.GroundException;
-import edu.berkeley.ground.plugins.hive.util.JsonUtil;
+import edu.berkeley.ground.plugins.hive.util.PluginUtil;
 
 public class GroundPartition {
 
-  static final private Logger LOG = LoggerFactory.getLogger(GroundTable.class.getName());
+    static final private Logger LOG = LoggerFactory.getLogger(GroundTable.class.getName());
 
-  private static final String DUMMY_NOT_USED = "DUMMY_NOT_USED";
-  private static final List<String> EMPTY_PARENT_LIST = new ArrayList<String>();
+    private static final long DUMMY_NOT_USED = 1L;
+    private GroundReadWrite groundReadWrite = null;
 
-  private GroundReadWrite groundReadWrite = null;
-
-  public GroundPartition(GroundReadWrite ground) {
-    groundReadWrite = ground;
-  }
-
-  public Node getNode(String partitionName) throws GroundException {
-    try {
-      LOG.debug("Fetching partition node: " + partitionName);
-      return groundReadWrite.getNodeFactory().retrieveFromDatabase(partitionName);
-    } catch (GroundException ge1) {
-      LOG.debug("Not found - Creating partition node: {}", partitionName);
-
-      Node node = groundReadWrite.getNodeFactory().create(partitionName, new HashMap<>());
-      Structure nodeStruct = groundReadWrite.getStructureFactory().create(node.getName(), new HashMap<>());
-
-      return node;
+    public GroundPartition(GroundReadWrite ground) {
+        groundReadWrite = ground;
     }
-  }
 
-  public Structure getNodeStructure(String partitionName) throws GroundException {
-    try {
-      Node node = this.getNode(partitionName);
-      return groundReadWrite.getStructureFactory().retrieveFromDatabase(partitionName);
-    } catch (GroundException e) {
-      LOG.error("Unable to fetch parition node structure");
-      throw e;
+    public Node getNode(String partitionName) throws GroundException {
+        LOG.debug("Fetching partition node: " + partitionName);
+        return groundReadWrite.getGroundReadWriteNodeResource().getNode(partitionName);
     }
-  }
-
-  public Edge getEdge(String partitionName) throws GroundException {
-    try {
-      LOG.debug("Fetching table partition edge: " + partitionName);
-      return groundReadWrite.getEdgeFactory().retrieveFromDatabase(partitionName);
-    } catch (GroundException ge1) {
-      LOG.debug("Not found - Creating table partition edge: {}", partitionName);
 
-      Edge edge = groundReadWrite.getEdgeFactory().create(partitionName, new HashMap<>());
-      Structure edgeStruct = groundReadWrite.getStructureFactory().create(partitionName, new HashMap<>());
-      return edge;
+    public Structure getNodeStructure(String partitionName) throws GroundException {
+        try {
+            Node node = this.getNode(partitionName);
+            return groundReadWrite.getGroundReadWriteStructureResource().getStructure(node.getName());
+        } catch (GroundException e) {
+            LOG.error("Unable to fetch parition node structure");
```

Comment:
As both `getNode` and `getStructure` could throw a `GroundException`, I don't think we can assert that the error here is due to fetching the partition. Either add a second `try` for the `getStructure` call or remove altogether and just let the exception surface.  (I'm sporadically visiting the project, so choose based on project current style).
There's also a typo in 'par**t**ition'.